examples/armv4t/gdb/host_io.rs

```diff
@@ -262,7 +262,11 @@ impl target::ext::host_io::HostIoReadlink for Emu {
             .to_str()
             .ok_or(HostIoError::Errno(HostIoErrno::ENOENT))?
             .as_bytes();
-        Ok(copy_range_to_buf(data, 0, data.len(), buf))
+        if data.len() <= buf.len() {
```

Comment:
this is fine... but you didn't add the this logic to the hard-coded file names above.

yes, I _know_ that it's incredibly unlikely that the buffer wouldn't be able to contain those strings, but the point of example code is to show off what a reasonably correct implementation looks like.

Please update the `copy_range_to_buf` blocks above with the same guards.

Reply:
I didn't add it because RSP packet has the minimum size:
>  If this stub feature is not supported, GDB guesses based on the size of the ‘g’ packet response.

Reply:
bahhh, you're probably right. on second thought, this seems fiiiiiiine